go/vt/vtgate/semantics/derived_table.go

```diff
@@ -66,7 +68,7 @@ func createDerivedTableForExpressions(expressions sqlparser.SelectExprs, cols sq
 func (dt *DerivedTable) dependencies(colName string, org originable) (dependencies, error) {
 	directDeps := org.tableSetFor(dt.ASTNode)
 	for i, name := range dt.columnNames {
-		if name != colName {
+		if !strings.EqualFold(name, colName) {
```

Comment:
Is this fixing a potential bug / does this need to be backported? 🤔

Reply:
It fixes https://github.com/vitessio/vitess/issues/11942

Yeah, we could backport just this change. We already have a test that exposes the problem in the plan tests. That's how I found it